tests/components/mqtt_room/test_sensor.py

```diff
@@ -65,20 +100,34 @@ async def test_room_update(hass, mqtt_mock_entry_with_yaml_config):
     )
     await hass.async_block_till_done()
 
-    await send_message(hass, BEDROOM_TOPIC, FAR_MESSAGE)
-    await assert_state(hass, BEDROOM)
-    await assert_distance(hass, 10)
+    await assert_room_update(hass)
 
-    await send_message(hass, LIVING_ROOM_TOPIC, NEAR_MESSAGE)
-    await assert_state(hass, LIVING_ROOM)
-    await assert_distance(hass, 1)
 
-    await send_message(hass, BEDROOM_TOPIC, FAR_MESSAGE)
-    await assert_state(hass, LIVING_ROOM)
-    await assert_distance(hass, 1)
+async def test_room_update_with_extra_rooms(hass, mqtt_mock_entry_with_yaml_config):
+    """Test the updating between rooms with extra rooms activated."""
+    assert await async_setup_component(
+        hass,
+        sensor.DOMAIN,
+        {
+            sensor.DOMAIN: {
+                CONF_PLATFORM: "mqtt_room",
+                CONF_NAME: NAME,
+                CONF_DEVICE_ID: DEVICE_ID,
+                CONF_STATE_TOPIC: "room_presence",
+                CONF_QOS: DEFAULT_QOS,
+                CONF_TIMEOUT: 5,
+                CONF_EXTRA_ROOMS: 2,
```

Comment:
What if this setting is something else (e.g. 100) will it fail?

Reply:
I can add another test-case that effectively reruns the same test, just a higher configuration value.